TASK: Run checks against current database versions

kdambekalns opened the pull request:
This change

- updates the MariaDB version used for testing from 10.2 to 10.6. The former reaches EOL in May 2022 but 10.6 is the current LTS version supported until July 2026
- updates the PostgreSQL version used for testing from 9.5 (reached EOL in February 2021) to 10.x, the oldest still supported release
- adds a nighty test against all supported PostgreSQL versions (10.x to 14.x)

kdambekalns committed:
TASK: Add build to check against PostgreSQL versions

kdambekalns committed:
TASK: Run tests against current PostgreSQL version

kdambekalns committed:
TASK: Run tests against current MariaDB version

Version 10.2 is supported only until May 2022, which is soon.

kdambekalns commented:
The new PostgreSQL tests can be seen in action at https://github.com/kdambekalns/flow-development-collection/actions/runs/2095298897

kdambekalns commented:
@mficzel This could go into 8.0, if you agree. I'd then change the target branch.

albe commented on .github/workflows/postgresql-versions.yml:
```diff
@@ -0,0 +1,145 @@
+        # Use https://hub.docker.com/_/postgres for available versions
+        postgresql-versions: ['14-alpine', '13-alpine', '12-alpine', '11-alpine', '10-alpine']
+        dependencies: ['highest']
+        composer-arguments: ['--ignore-platform-reqs'] # to run --ignore-platform-reqs in experimental builds
```
I think the `--ignore-platform-reqs` isn't required here. It's to allow installing composer dependencies that are prohibited for the given PHP version, so mostly useful for nightly/latest tests

albe replied:
```suggestion
        composer-arguments: [''] # to run --ignore-platform-reqs in experimental builds
```

mficzel commented:
Fine with targeting 8.0 for that. Makes sense me but habe no time for a proper review myself ATM

albe commented:
> @mficzel This could go into 8.0, if you agree. I'd then change the target branch.

Since this is a scheduled workflow and the target branch to check against is chosen in the matrix, it won't make a difference if this exists in master only.

kdambekalns committed:
TASK: Apply review suggestion

Co-authored-by: Alexander Berl <a.berl@outlook.com>

kdambekalns commented:
> Since this is a scheduled workflow and the target branch to check against is chosen in the matrix, it won't make a difference if this exists in master only. More concretely, it even must exist in master so that the workflow actually runs.

True. But the updates to `build.yml` (for the "regular CI") might be something for 8.0…

mficzel commented:
Would it make sense to run the deps "lowest" tests using the oldest reasonable db version and deps "highest" the newest.
That way we would get immediate results, offcourse without any informations about the versions in between.

I fear a nightly job is very easy to ignore

kdambekalns commented:
> Would it make sense to run the deps "lowest" tests using the oldest reasonable db version and deps "highest" the newest. That way we would get immediate results, offcourse without any informations about the versions in between.

Yes, I created  a new PR against 8.0 that does that, more or less. We limit the matrix a bit and per PHP version run only one MariaDB and one PostgreSQL version. But I adjusted the build, so that the "lowest dependencies" job runs against the "lowest database version".

See https://github.com/neos/flow-development-collection/pull/2825

> I fear a nightly job is very easy to ignore

Might be, but still better than nothing. I think running against all DB versions is too resource-hungry and time consumig.

kdambekalns committed:
TASK: Use postgres as service 

Improved based on https://github.com/neos/neos-development-collection/pull/3695

skurfuerst reviewed (approved):
Awesome :)

albe reviewed (approved):
👍
